src/pages/index.js

```diff
@@ -124,23 +161,109 @@ function handleEscapeKey(evt) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editProfileModalNameInput.value;
-  profileDescription.textContent = editProfileModalDescriptionInput.value;
-  closeModal(editProfileModal);
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+  api
+    .editUserInfo({
+      name: editProfileModalNameInput.value,
+      about: editProfileModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editProfileModal);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+
   evt.target.reset();
 }
 
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
-  const inputValues = {
-    name: cardModalNameInput.value,
-    link: cardModalLinkInput.value,
-  };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  closeModal(editCardModal);
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+  api
+    .addNewCard({
+      name: cardModalNameInput.value,
+      link: cardModalLinkInput.value,
+    })
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      closeModal(editCardModal);
+      evt.target.reset();
+      disableButton(cardSubmitBtn);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+}
+
+function handleDeleteCardSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Deleting...";
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deteleCardModal);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitBtn.textContent = "Delete";
+    });
+}
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deteleCardModal);
+}
+
+function handleAvatarFormSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+
+  api
+    .editAvatar({
+      avatar: avatarInput.value,
+    })
+    .then((data) => {
+      document.querySelector(".profile__picture").src = data.avatar;
+      closeModal(avatarModal);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+
   evt.target.reset();
-  disableButton(cardSubmitBtn, settings);
+  disableButton(avatarSubmitBtn);
```

Comment:
```js

  evt.target.reset();
  disableButton(avatarSubmitBtn);
```

These 2 lines should be in `.then` in line 246

* You need to **reset** the form only   in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.
* You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.